.github/workflows/lint.yml

```
@@ -52,10 +52,10 @@ jobs:
         working-directory: OpenSearch-Dashboards/plugins/${{ env.PLUGIN_NAME }}
         run: |
           git fetch origin main
-          CHANGED_FILES=$(git diff --name-only --diff-filter=ACMRTUXB origin/main | grep -E "\.(js|ts|tsx)$")
+          CHANGED_FILES=($(git diff --name-only --diff-filter=d origin/main | grep -E "\.(js|ts|tsx)$")) || true
           if [ -n "$CHANGED_FILES" ]; then
             echo "Linting changed files..."
-            yarn lint $CHANGED_FILES
+            yarn lint "${CHANGED_FILES[@]}"
           else
-            echo "No JavaScript/TypeScript files changed."
+            echo "No matching files changed."
           fi
```

```suggestion
      - name: Get list of changed files
        id: files
        run: |
          BASE_SHA="${{ github.event.pull_request.base.sha }}"
          HEAD_SHA="${{ github.event.pull_request.head.sha }}"
          git fetch origin $BASE_SHA
          git diff --name-only $BASE_SHA...$HEAD_SHA > changed_files.txt
          CHANGED_FILES=$(cat changed_files.txt | grep -E '\.(js|ts|tsx)$' || true)
          echo "::set-output name=changed::${CHANGED_FILES}"
        working-directory: OpenSearch-Dashboards/plugins/${{ env.PLUGIN_NAME }}

      - name: Lint Changed Files
        run: |
          CHANGED_FILES="${{ steps.files.outputs.changed }}"
          if [[ -n "$CHANGED_FILES" && "$CHANGED_FILES" != "" ]]; then
            echo "Linting changed files..."
            IFS=$'\n' read -r -a FILES_TO_LINT <<< "$CHANGED_FILES"
            yarn lint "${FILES_TO_LINT[@]}"
          else
            echo "No JavaScript/TypeScript files to lint."
          fi
        working-directory: OpenSearch-Dashboards/plugins/${{ env.PLUGIN_NAME }}
```
Don't forget to change the name at L51

---

Thanks for suggestions. 
Note: Tested and synced with @RyanL1997, still yet to make it working in all test cases, for now just go ahead with hardcoding main instead of using sha. Will follow up on this.

---

nit. the `&& "$CHANGED_FILES" != ""` seems redundant?